Update "Explicit MFA Deny"

tduarte14 opened the pull request:
Required items, please complete
   
   Change(s):
   - Remove redundant line declaring Type (it's already there)
   - Add Host and OS fields to the end of the query
   - Add extra entities: App and Host
   - Fix IP entity (was using inexistent field)
   - Removed URL entity (was using ClientAppUsed, which was giving no value)

   Reason for Change(s):
   - Some Entities had broken field mapping, another had no value and some were needed.

   Version Updated:
   - Yes

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

ashwin-patil commented on Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml:
```diff
@@ -49,14 +49,20 @@ entityMappings:
-  - entityType: URL
+      - identifier: ResourceId
+        columnName: ResourceId
+  - entityType: CloudApplication
```
can you also add  identifier: AppId and Fieldmapping for CloudApplication entity.

ashwin-patil reviewed (changes requested):
suggested couple of changes on entity mappings

tduarte14 commented on Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml:
```diff
@@ -24,21 +24,21 @@ relevantTechniques:
   | project-rename PublicIP = IPAddress   
-  | join kind=leftouter dvcInfo on PublicIP  
+  | join kind=leftouter dvcInfo on PublicIP
+  | extend Host=tostring(DeviceDetail_dynamic.displayName), OS=tostring(DeviceDetail_dynamic.operatingSystem)
```
The thing is that the same Public IP can be related to different devices, so I'm not sure you can even have a correct match from that block of code there. The host I got was from the signin logs, which is what we actually see on the logs. DeviceInfo shows many times no publicIP nor logged user, so it's hard to have a match.

tduarte14 replied:
If you please don't agree, please review again.

ashwin-patil replied:
Sounds good @tduarte14 . We will revisit this sometime again later. 
Thanks for the feedback.

tduarte14 reviewed (commented):
added comments

tduarte14 reviewed (commented):
added comments as why I don't agree on the review. Please check

tduarte14 commented:
Hi any feedback here?

v-prasadboke commented:
Hello @tduarte14 Ashwin is on leave till 22. He will get back to you as he comes back.

ashwin-patil commented:
> Hi any news here?

Hey, i started the review but forgot to submit final comments. Apologies for delay, let me know your thoughts on follow-up comments. I will keep an eye on this PR this week to respond sooner.

v-prasadboke commented:
Hey Thank you for the updates @ashwin-patil.

v-prasadboke commented:
Hello @tduarte14, Ashwin has requested some changes please work on it

v-prasadboke commented:
Hello @tduarte14,
Ashwin has suggested some changes please take a look into it.

v-prasadboke commented:
Hello @tduarte14, 
Ashwin has requested some changes earlier, Please have a look on it.

v-prasadboke commented:
Hello @tduarte14, 
You might be busy with something, commenting on this PR to please acknowledge Ashwin's comments.

tduarte14 commented:
Got the latest version 1.0.5, removed URL entity and added just CloudApplication entity. Ignored the Host entity for now

tduarte14 commented:
Also changed runtime to every 1h instead of 1d to improve MTTD. Added missing appId field on CloudApplication entity as you requested, thought it was there already

v-prasadboke commented:
Thank you @tduarte14.
